config/src/main/java/org/springframework/security/config/annotation/web/configurers/HeadersConfigurer.java

```diff
@@ -658,6 +656,79 @@ private HpkpConfig enable() {
 	}
 
 	/**
+	 * <p>
+	 * Allows configuration for <a href="https://www.w3.org/TR/CSP2/">Content Security Policy (CSP) Level 2</a>.
+	 * </p>
+	 *
+	 * <p>
+	 * Calling this method automatically enables (includes) the Content-Security-Policy header in the response
+	 * using the supplied Security Policy Directive(s).
+	 * </p>
+	 *
+	 * <p>
+	 * Configuration is provided to the {@link ContentSecurityPolicyHeaderWriter} which supports the writing
+	 * of the two headers as detailed in the specification:
+	 * <ul>
+	 * 	<li>Content-Security-Policy</li>
+	 * 	<li>Content-Security-Policy-Report-Only</li>
+	 * </ul>
+	 * </p>
+	 *
+	 * @see ContentSecurityPolicyHeaderWriter
+	 * @since 4.1
+	 * @return the ContentSecurityPolicyConfig for additional configuration
+	 */
+	public ContentSecurityPolicyConfig contentSecurityPolicy(String policyDirectives) {
+		return contentSecurityPolicy.directives(policyDirectives);
+	}
+
+	public final class ContentSecurityPolicyConfig {
+		private ContentSecurityPolicyHeaderWriter writer;
+
+		private ContentSecurityPolicyConfig() {
+		}
+
+		/**
+		 * Sets the supplied security policy directive(s) in the response.
+		 *
+		 * @param policyDirectives the security policy directive(s) to write in the response.
+		 *
+		 * @return the {@link ContentSecurityPolicyConfig} for additional configuration
+		 */
+		public ContentSecurityPolicyConfig directives(String policyDirectives) {
+			if (writer == null) {
+				this.writer = new ContentSecurityPolicyHeaderWriter(policyDirectives);
+			} else {
+				// TODO Allow for over-riding previously set directives? Is this really needed?
+				this.writer.setPolicyDirectives(policyDirectives);
```

Comment:
I think this should be removed. User's can always invoke `contentSecurityPolicy` again.